cranelift/filetests/filetests/runtests/simd-swizzle.clif

```diff
@@ -0,0 +1,33 @@
+test interpret
+test run
+target aarch64
+set enable_simd
+target x86_64
+
+function %swizzle_i8x16(i8x16, i8x16) -> i8x16 {
+block0(v0: i8x16, v1: i8x16):
+    v2 = swizzle.i8x16 v0, v1
+    return v2
+}
+; run: %swizzle_i8x16([1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16], [0 9 15 1 6 13 7 11 10 8 100 12 4 2 3 5]) == [1 10 16 2 7 14 8 12 11 9 0 13 5 3 4 6]
+
+function %swizzle_i16x8(i8x16, i8x16) -> i16x8 {
+block0(v0: i8x16, v1: i8x16):
+    v2 = swizzle.i16x8 v0, v1
```

Comment:
I'm slightly surprised that I allowed swizzles for types other than `i8x16` ([here](https://github.com/bytecodealliance/wasmtime/blob/main/cranelift/codegen/meta/src/shared/instructions.rs#L546-L561)). In the spirit of constraining the instruction set, this might be another place to consider pruning: it is a bit strange that one can send in an `i8x16`, rearrange it with the swizzle mask, and then have some other vector type. Any thoughts on this? @cfallin, @akirilov-arm, @jlb6740?

(In any case, this PR should merge as-is and we can remove these later if that's what we decide--thanks for finding this!).

Reply:
Yes, in principle I agree; I think we should be sort of systematic about this in a pass over all instructions, thinking about what types make sense, so I don't think we need to make an issue just for this, or adjust the test right now, but noted!